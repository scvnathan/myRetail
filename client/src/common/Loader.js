import React from "react";
import logo from "~images/logo.png";
import smallLogo from "~images/logo-sm.png";
import styled, { keyframes } from "styled-components";

const loaderAnim = keyframes`
	0%,47%, 100% { transform: scale3d(1,1,1); }
	10% { transform: scale3d(1.25,.75,1); }
	14% { transform: scale3d(.85,1.15,1); }
	20% { transform: scale3d(1.15,.85,1); }
	24% { transform: scale3d(.95,1.05,1); }
	36% { transform: scale3d(1.05,.95,1); }
`;

const LoaderLogo = styled.img`
	animation: ${loaderAnim};
	animation-duration: 2s;
	animation-timing-function: ease;
	animation-delay: 0s;
	animation-iteration-count: infinite;
	animation-direction: normal;
	animation-fill-mode: both;
`;

const Wrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	margin: 0 auto;
	height: 100vh;
	justify-content: center;
	width: 45%;
`;

export default () => {
	return <LoaderLogo src={smallLogo} />;
};

export function FullScreenLoader() {
	return (
		<Wrapper>
			<LoaderLogo src={logo} />
		</Wrapper>
	);
}
