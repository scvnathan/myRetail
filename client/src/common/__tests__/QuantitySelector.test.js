import React from "react";
import Loader from "../QuantitySelect";
import renderer from "react-test-renderer";
import {shallow} from "enzyme"
import QualitySelect from "~root/common/QuantitySelect"

describe("QuantitySelect component", () => {
	let wrapper;
	const quantityValueSel = '[data-test="quantityValue"]';
	const incSel = '[data-test="incBtn"]';
	const decSel = '[data-test="decBtn"]';
	let getQuantity = (w) => parseInt(w.find(quantityValueSel).text(), 10);

	beforeEach(() => {
		wrapper = shallow(<QualitySelect />);
	})

	it("should render correctly", () => {
		const comp = <Loader />;
		const tree = renderer.create(comp).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should increment the local count", () => {
		const inc = wrapper.find(incSel)
		const originalValue = getQuantity(wrapper);

		inc.simulate('click');
		expect(getQuantity(wrapper)).toBe(originalValue + 1);
	});

	it("should decrement the local count", () => {
		const inc = wrapper.find(incSel)
		const dec = wrapper.find(decSel)
		const originalValue = getQuantity(wrapper);

		inc.simulate('click');
		expect(getQuantity(wrapper)).toBe(originalValue + 1);

		dec.simulate('click');
		expect(getQuantity(wrapper)).toBe(originalValue);
	});
});
