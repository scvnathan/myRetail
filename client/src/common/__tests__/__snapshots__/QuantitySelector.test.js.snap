// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantitySelect component should render correctly 1`] = `
<div
  className="QuantitySelect__Selector-sc-1732u5f-0 VPRYu"
>
  quantity:
  <div
    className="QuantitySelect__ButtonRow-sc-1732u5f-1 drZiNT"
  >
    <div
      className="QuantitySelect__DecrementButton-sc-1732u5f-3 bJgkaZ"
      data-test="decBtn"
      onClick={[Function]}
    />
    <strong
      data-test="quantityValue"
      style={
        Object {
          "paddingTop": "5%",
        }
      }
    >
      1
    </strong>
    <div
      className="QuantitySelect__IncrementButton-sc-1732u5f-2 lgLktx"
      data-test="incBtn"
      onClick={[Function]}
    />
  </div>
</div>
`;
